test/testdrive/kafka-upsert-sources.td

```diff
@@ -257,6 +257,14 @@ mammal1:
   VALUE FORMAT TEXT
   ENVELOPE UPSERT
 
+! select * from nullkey
+contains: message with NULL key in partition 0
```

Comment:
@morsapaes & @sjwiesman any opinions about that error, and new behaviour?

Reply:
Is there anything the user can do to fix this besides drop the source? If yes, maybe we include that here or in the docs. If not +1

Reply:
@sjwiesman if you send a value with a null key and value, the source comes back to life

Reply:
LGTM, but I'd +1 including a hint with the fix (which would be awkward to document).

Reply:
I had no point of reference here, so checked how Kafka handles `NULL` keys passed to compacted topics and it also throws a non-retriable "Invalid key" exception ([KIP-135](https://cwiki.apache.org/confluence/display/KAFKA/KIP-135+%3A+Send+of+null+key+to+a+compacted+topic+should+throw+non-retriable+error+back+to+user)).

Reply:
@morsapaes ty! that is valuable context.

I'll also add a hint about how to retract to the error message.